### PR TITLE
fix: bug preventing str-type when specifying transaction types.

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Fixes regression where we were not longer able to specify `type="0x0"` for static fee transactions.
Adds Transactions guide.

### How I did it

handle hexbytes differently than bytes

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: docs/userguides/transactions.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def deploy():
+    account = accounts.load("MyAccount")
+    return account.deploy(project.MyContract)
+```
+
+## Dynamic Fee Transactions
+
+Before [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)), all transaction used a `gas_price`. After the London fork
```
Comment (unparalleled-js) [inline docs/userguides/transactions.md:26]:
```suggestion
Before [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)), all transactions used a `gas_price`. After the London fork
```

File: docs/userguides/transactions.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Calling certain methods on a deployed-contract is a way to transact.
+
+```python
+contract = deploy()  # Example from above, that returns a contract instance.
+contract.fundMyContract(value="1 gwei")  # Assuming there is a method named 'fundMyContract' on MyContract.
+```
+
+In the example above, the call to `fundMyContract()` invokes a dynamic-fee transaction. To have more control of the 
+fee-values, you can specify the `max_fee` and the `max_priority_fee`.
```
Comment (fubuloubu) [inline docs/userguides/transactions.md:39]: you can do just one or the other too

File: docs/userguides/transactions.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+```python
+contract = deploy()  # Example from above, that returns a contract instance.
+contract.fundMyContract(value="1 gwei", max_priorty_fee=50000000000, max_fee=100000000000)
```
Comment (fubuloubu) [inline docs/userguides/transactions.md:43]: This should work now too:
```suggestion
contract.fundMyContract(value="1 gwei", max_priority_fee="50 gwei", max_fee="1.2 gwei")
```

File: docs/userguides/transactions.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+## Static Fee Transactions
+
+To use the original static fee transactions, you can merely specify a `type="0x0"`.
```
Comment (fubuloubu) [inline docs/userguides/transactions.md:48]:
```suggestion
To use the original static fee transactions without having to specify `gas_price`, you just have to specify `type="0x0"`.
```

File: docs/userguides/transactions.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+```python
+contract = deploy()  # Example from above, that returns a contract instance.
+contract.fundMyContract(value="1 gwei", type="0x0")
+```
+
+Also, specifying a `gas_price` will automatically trigger the usage of a static-fee transaction.
+
+```python
+contract = deploy()  # Example from above, that returns a contract instance.
+contract.fundMyContract(value="1 gwei", gas_price=100000000000)
```
Comment (fubuloubu) [inline docs/userguides/transactions.md:59]:
```suggestion
contract.fundMyContract(value="1 gwei", gas_price="100 gwei")
```

File: docs/userguides/transactions.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+## Static Fee Transactions
+
+To use the original static fee transactions, you can merely specify a `type="0x0"`.
+
+```python
+contract = deploy()  # Example from above, that returns a contract instance.
+contract.fundMyContract(value="1 gwei", type="0x0")
+```
+
+Also, specifying a `gas_price` will automatically trigger the usage of a static-fee transaction.
```
Comment (fubuloubu) [inline docs/userguides/transactions.md:55]: This sentence being before the previous is a little confusing. One of them should change so it reads better.

**Review by fubuloubu — APPROVED** [review]

File: docs/userguides/transactions.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+def deploy():
+    account = accounts.load("MyAccount")
+    return account.deploy(project.MyContract)
+```
+
+## Dynamic Fee Transactions
+
+Before [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)), all transactions used a `gas_price`. After the London fork
+of Etheruem, the `gas_price` got broken up into two values, `max_fee` and `max_priority_fee`. The `ape` platform
```
Comment (fubuloubu) [inline docs/userguides/transactions.md:27]:
```suggestion
of Etheruem, the `gas_price` got broken up into two values, `max_fee` and `max_priority_fee`. The `ape` framework
```

Comment (fubuloubu) [reply docs/userguides/transactions.md:27]: Also, I usually put different sentences on new lines. Markdown will shove them together.

File: docs/userguides/transactions.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+contract.fundMyContract(value="1 gwei", gas_price="100 gwei")
+```
+
+Another way to use a static-fee transaction (without having to provide `gas_price`) is to set the key-value
+argument `type` equal to `0x00`.
+
+```python
+contract.fundMyContract(value="1 gwei", type="0x0")
+```
+
```
Comment (fubuloubu) [inline docs/userguides/transactions.md:59]: Add a note that `max_priority_fee` cannot exceed `max_fee`, and that `base_fee = max_fee - max_priority_fee`, which much match network conditions in order to be mined.

Lastly, what happens if `max_priority_fee` is omitted or `max_fee` is omitted in terms of the value of the unspecified fee?

I would think that specifying `max_priority_fee` means `max_fee = ProviderAPI.base_fee + max_priority_fee`, and that specifying `max_fee` without `max_priority_fee` means `max_priority_fee` will be some average over the last N blocks of that value, and `base_fee` is set to `max_fee - max_priority_fee` (and might take longer to confirm based on the current value of `base_fee`, which we can create a warning about)

**Review by fubuloubu — DISMISSED** [review]

File: docs/userguides/transactions.md (hunk 8 above)
Comment (fubuloubu) [reply docs/userguides/transactions.md:59]: Also I should specify that we have to do this lmao

Comment (unparalleled-js) [reply docs/userguides/transactions.md]: I think your expectations of what happens is correct. Here is where the defaults get set: https://github.com/ApeWorX/ape/blob/main/src/ape/api/accounts.py#L109-L115

* If you omit the `max_priority_fee` and and include the `max_fee`, the `max_priority_fee` just gets set from the provider.

* If you omit the `max_fee` and include the `max_priority_fee`, the `max_fee` still just becomes the `base_fee` plus the `max_priority_fee`.

As far as I know, there is no way to set the base fee. I don't know how `priority_fee` gets calculated by the provider.

File: docs/userguides/transactions.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+def deploy():
+    account = accounts.load("MyAccount")
+    return account.deploy(project.MyContract)
+```
+
+## Dynamic Fee Transactions
+
+Before [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559)), all transactions used a `gas_price`.
+After the London fork of Etheruem, the `gas_price` got broken up into two values, `max_fee` and `max_priority_fee`.
+The `ape` framework supports both types of transactions. By default, transactions use the dynamic fee model.
```
Comment (unparalleled-js) [inline docs/userguides/transactions.md:28]:
```suggestion
The `ape` framework supports both types of transactions. By default, transactions use the dynamic-fee model.
```

File: docs/userguides/transactions.md (hunk 8 above)
Comment (fubuloubu) [reply docs/userguides/transactions.md]: > As far as I know, there is no way to set the base fee.

That is correct, it is essentially a "protocol parameter" set by the previous block and how full/empty it is.

> And I don't know how `priority_fee` gets calculated by the provider.

The formula `max_fee = base_fee + priority_fee` is probably used.

File: docs/userguides/transactions.md [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+One way to use a static-fee transaction is by specifying the `gas_price` as a key-value argument:
+
+```python
+contract.fundMyContract(value="1 gwei", gas_price="100 gwei")
+```
+
+**NOTE**: Miners prioritize static-fee transactions based on the highest `gas_price`.
+
+Another way to use a static-fee transaction (without having to provide `gas_price`) is to set the key-value
+argument `type` equal to `0x00`.
```
Comment (fubuloubu) [inline docs/userguides/transactions.md:69]: Add "By default, the gas_price is set using the provider's estimation" or something similar.

**Review by fubuloubu — DISMISSED** [review]

File: docs/userguides/transactions.md (hunk 10 above)
Comment (unparalleled-js) [reply docs/userguides/transactions.md:69]: - [x] Done!

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**